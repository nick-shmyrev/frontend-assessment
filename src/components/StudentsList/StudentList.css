.students-list {
  background: #fff;
  border-radius: 10px;
  border: 1px solid #ccc;
  width: 80vw;
  height: 80vh;
  overflow-y: scroll;
  padding: 0;
  margin: 0;
}

::-webkit-scrollbar { display:none }
