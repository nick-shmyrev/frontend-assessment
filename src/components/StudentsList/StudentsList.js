import React from 'react';
import { connect } from 'react-redux';
import getVisibleStudents from './../../store/selectors/students-selectors';

// Import components
import StudentItem from '../StudentItem/StudentItem';

// Import styles
import style from './StudentList.css';

const StudentsList = ({ students }) => (
  <ul className={style['students-list']}>
    {
      students.length ?
        students.map(student => <StudentItem {...student} key={student.email}/>) :
        <p>No students data to display...</p>
    }
  </ul>
);

const mapStateToProps = ({ students, filters }) => ({
  students: getVisibleStudents(students, filters),
});

export default connect(mapStateToProps)(StudentsList);
