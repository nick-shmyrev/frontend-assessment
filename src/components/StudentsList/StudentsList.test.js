import React from 'react';
import { shallow } from 'enzyme';
import { StudentsList } from './StudentsList';

describe('StudentsList component', () => {
  const students = [{
    city: "FushÃ«-Muhurr",
    company: "Yadel",
    email: "iorton0@imdb.com",
    firstName: "Ingaberg",
    grades: [
      "78",
      "100",
      "92",
      "86",
      "89",
      "88",
      "91",
      "87"
    ],
    id: "1",
    lastName: "Orton",
    pic: "https://storage.googleapis.com/hatchways-app.appspot.com/assessments/data/frontend/images/voluptasdictablanditiis.jpg",
    skill: "Oracle"
  },
  {
    city: "Sanghan",
    company: "Avamm",
    email: "cboards1@weibo.com",
    firstName: "Clarke",
    grades: [
      "75",
      "89",
      "95",
      "93",
      "99",
      "82",
      "89",
      "76"
    ],
    id: "2",
    lastName: "Boards",
    pic: "https://storage.googleapis.com/hatchways-app.appspot.com/assessments/data/frontend/images/voluptasautreprehenderit.jpg",
    skill: "Sports"
  },
  ];
  
  test('should render without students', () => {
    const wrapper = shallow(<StudentsList students={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should render with students provided', () => {
    const wrapper = shallow(<StudentsList students={students}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
