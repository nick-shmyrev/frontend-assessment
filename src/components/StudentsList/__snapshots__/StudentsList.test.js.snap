// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentsList component should render with students provided 1`] = `
<ul
  className="students-list"
>
  <Connect(StudentItem)
    city="FushÃ«-Muhurr"
    company="Yadel"
    email="iorton0@imdb.com"
    firstName="Ingaberg"
    grades={
      Array [
        "78",
        "100",
        "92",
        "86",
        "89",
        "88",
        "91",
        "87",
      ]
    }
    id="1"
    key="iorton0@imdb.com"
    lastName="Orton"
    pic="https://storage.googleapis.com/hatchways-app.appspot.com/assessments/data/frontend/images/voluptasdictablanditiis.jpg"
    skill="Oracle"
  />
  <Connect(StudentItem)
    city="Sanghan"
    company="Avamm"
    email="cboards1@weibo.com"
    firstName="Clarke"
    grades={
      Array [
        "75",
        "89",
        "95",
        "93",
        "99",
        "82",
        "89",
        "76",
      ]
    }
    id="2"
    key="cboards1@weibo.com"
    lastName="Boards"
    pic="https://storage.googleapis.com/hatchways-app.appspot.com/assessments/data/frontend/images/voluptasautreprehenderit.jpg"
    skill="Sports"
  />
</ul>
`;

exports[`StudentsList component should render without students 1`] = `
<ul
  className="students-list"
>
  <p>
    No students data to display...
  </p>
</ul>
`;
